### DISCUSSION TITLE
Something went wrong during the adapter installation

### DISCUSSION THREAD

**Discussion by Bauaser-kun** [apache/streampipes#2498] [Q&A] [answered]
Hello
I've gotten rather mysterious error message while creating new adapter of any kind. (I tried to many different and all are returning the same message).
Initialy the adapter was suppoused to connect to VM by IP (kafka adapter) and I got to load schema properly from topic, the message occurs on last step.
This occours on our development envireonment, on custom-made environment (streampipes env --set streampipes-element-custom)
the file setting this environment looks like this.
Have anyone faced similiar issue?

```
[environment:pipeline-element-custom]  
backend  
consul  
couchdb  
kafka  
ui  
zookeeper  
influxdb  
extensions-all-jvm
```

I use Apache streampipes 0.93.

![image](https://github.com/apache/streampipes/assets/72672258/1dc35a81-1c11-4abf-b2bc-12538a6c2699)

**Comment by tenthe** [comment]
Hi @Bauaser-kun,
Sorry, I didn't have this problem before. But I have some questions to find the cause of the problem:

- Are the icons of the adapters and processing elements displayed correctly?
- Can you check the logs of the services `backend` and `extensions-all-jvm`? Have these services been started without exception? Is there an exception when you try to start the adapter?

Cheers,
Philipp

**Reply by Bauaser-kun** [reply]
Sorry for the late reply.
All icons are showing properly both in adapter creation and pipelines views.
 I've checked logs. The extension-all-jvm log ends on february 29th approximetely the time I first time tried to create adapter, the last log is: 

`Error grabbing logs: invalid character '\x00' looking for beginning of value`

The backend log for today's streampipe startup is working but it indeed shows some errors from the very beggining - even though all containers seem to be up properly while they are initialized i find the following error :

```backend_1             | 2024-03-04T07:50:13.275Z  INFO 1 --- [pool-1-thread-1] o.a.s.c.m.management.WorkerRestClient    : Trying to start adapter on endpoint http://172.31.0.7:8090/api/v1/worker/stream/invoke 
backend_1             | 2024-03-04T07:50:13.789Z ERROR 1 --- [pool-1-thread-1] o.a.s.c.m.management.WorkerRestClient    : Adapter was not started successfully
backend_1             | 
backend_1             | org.apache.http.conn.HttpHostConnectException: Connect to 172.31.0.7:8090 [/172.31.0.7] failed: Connection refused```

I assume that this is the OOTB adapter as it was definitely not created by me neither anyone else at my company.
But after that the same error occours for every single adapter we have already created in our pipelines. Then after around 15 seconds seconds the adapters are starting again, this time succesfully.

And then when I try to create new adapter I got this

```backend_1             | 2024-03-04T08:25:17.925Z  INFO 1 --- [nio-8030-exec-5] o.a.s.rest.impl.connect.AdapterResource  : User: admin@streampipes.apache.org starts adapter sp:org.apache.streampipes.connect.iiot.protocol.stream.kafka
backend_1             | 2024-03-04T08:25:18.092Z  WARN 1 --- [nio-8030-exec-5] o.g.j.server.ServerRuntime$Responder     : An exception mapping did not successfully produce and processed a response. Logging the exception propagated to the default exception mapper.
backend_1             | 
backend_1             | java.lang.NullPointerException: Cannot invoke "org.apache.streampipes.model.configuration.MessagingSettings.getPrioritizedProtocols()" because "messagingSettings" is null
backend_1             | 	at org.apache.streampipes.connect.management.util.GroundingUtils.createEventGrounding(GroundingUtils.java:50)
(...)```

**Reply by dominikriemer** [reply, marked as answer]
Hi @Bauaser-kun thanks for providing the logs.
That's a rather strange error so here are just some initial thoughts:
* In 0.93.0, we eliminated Consul and configurations such as the `messagingSettings` which are null according to the logs are now provided directly from a CouchDB document.
* The config is written either at installation time or when upgrading the system.
* Did you start from an empty installation or did you upgrade from a previous version? If yes, which was the version you had installed before doing the upgrade?
* It would be interesting to check the couchdb (you can log in to the web interface at `http://COUCHDB_HOST:5984/_utils`). Is there a database called `general-configuration` and document `core` in the database? If yes, does it include the messaging settings?
